spel/scripts/amigen7-build.sh

```diff
@@ -154,14 +155,26 @@ fi
 echo "Executing ChrootBuild.sh"
 bash -x "${ELBUILD}"/ChrootBuild.sh ${CLIOPT_CUSTOMREPO} ${CLIOPT_EXTRARPMS}
 
-# Epel mirrors are maddening; retry 5 times to work around issues
-echo "Executing AWScliSetup.sh"
-retry 5 bash -x "${ELBUILD}"/AWScliSetup.sh "${AWSCLISOURCE}" "${EPELRELEASE}" "${EPELREPO}"
+if [[ "${SPEL_CLOUDPROVIDER}" == "aws" ]]
+then
+    # Epel mirrors are maddening; retry 5 times to work around issues
+    echo "Executing AWScliSetup.sh"
+    retry 5 bash -x "${ELBUILD}"/AWScliSetup.sh "${AWSCLISOURCE}" "${EPELRELEASE}" "${EPELREPO}"
+fi
 
 echo "Executing ChrootCfg.sh"
 bash -x "${ELBUILD}"/ChrootCfg.sh
 
 echo "Executing GrubSetup.sh"
+if [[ "${SPEL_CLOUDPROVIDER}" == "azure" ]]
+then
+    #adding Azure grub defaults per https://docs.microsoft.com/en-us/azure/virtual-machines/linux/create-upload-centos#centos-70
+    /usr/bin/sed -i 's|rhgb|rootdelay=300|' "${ELBUILD}"/GrubSetup.sh
```

Comment:
As these are all operating on the same file, it might be a little more efficient to combine them into a single sed execution:

```
/usr/bin/sed -i \
    -e 's|rhgb|rootdelay=300|' \
    -e 's|quiet|earlyprintk=ttyS0|' \
    -e 's|crashkernel=auto ||' \
    -e 's|console=tty0|console=ttyS0|' \
    "${ELBUILD}"/GrubSetup.sh
```

Reply:
Usual scripted `sed` style would be:

~~~
/usr/bin/sed -i '{
   s|rhgb|rootdelay=300|
   s|quiet|earlyprintk=ttyS0|
   s|crashkernel=auto ||
   s|console=tty0|console=ttyS0|
}' "${ELBUILD}"/GrubSetup.sh
~~~

(less having to ass around with tics and backslashes)

Reply:
I found some circumstances where that syntax just doesn't seem to work, so I stick with multiple expressions. :/